pkg/tools/etcd_helper.go

```diff
@@ -354,6 +374,17 @@ func (h *EtcdHelper) AtomicUpdate(key string, ptrToType runtime.Object, ignoreNo
 	}
 }
 
+// Examines the etcd key and prepends the prefix if it isn't already there.
+func (h *EtcdHelper) prefixEtcdKey(key string) string {
```

Comment:
I am a little confused by this.

Do we pass around fully-qualified keys across methods, or are we always passing around the relative key path?

If I set my prefix to be "/registry/pods", then I would expect every key path in my repository to have that prefix prepended.  Instead, this behavior results in saying that my pods will go in "/registry/pods/default", but my services will go in "/registry/pods/registry/services/..." which basically breaks usage of a namespace called "registry" in this example.

In general, I think we should always prefix the key so in my example:

pods: /registry/pods/registry/pods/default/...
services: /registry/pods/registry/services/default/...

Right now, you are basically making the default prefix = "/registry" when you set it as "" as a result.